internal/resources/providers/awslib/config.go

```diff
@@ -18,30 +18,28 @@
 package awslib
 
 import (
-	"context"
+	"net/http"
 
-	awssdk "github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	libbeataws "github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
 )
 
-type ConfigProviderAPI interface {
-	InitializeAWSConfig(ctx context.Context, cfg aws.ConfigAWS) (*awssdk.Config, error)
-}
-
-type ConfigProvider struct {
-	MetadataProvider MetadataProvider
-}
-
-func (p ConfigProvider) InitializeAWSConfig(ctx context.Context, cfg aws.ConfigAWS) (*awssdk.Config, error) {
-	awsConfig, err := aws.InitializeAWSConfig(cfg)
+func InitializeAWSConfig(cfg libbeataws.ConfigAWS) (*aws.Config, error) {
+	awsConfig, err := libbeataws.InitializeAWSConfig(cfg)
 	if err != nil {
 		return nil, err
 	}
-	metadata, err := p.MetadataProvider.GetMetadata(ctx, awsConfig)
-	if err != nil {
-		return nil, err
+
+	awsConfig.Retryer = func() aws.Retryer {
+		return retry.NewStandard(func(o *retry.StandardOptions) {
+			o.Retryables = append(o.Retryables, retry.RetryableHTTPStatusCode{
+				Codes: map[int]struct{}{
+					http.StatusTooManyRequests: {},
+				},
+			})
+		})
 	}
-	awsConfig.Region = metadata.Region
```

Comment:
all our AWS config initializations are done via libbeat. the only extra thing the config provider did was set the region. it was only used by EKS, which now sets the region [itself](https://github.com/elastic/cloudbeat/pull/2223/files#diff-f297d6b58da94a9a8f8e33eed64ebdc64076afdf856ccd3d81e52e4b08225734R118)